lib/metrics/gc.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+const semver = require('semver');
+const Counter = require('../counter');
+const Summary = require('../summary');
+
+let perf_hooks;
+
+try {
+	// eslint-disable-next-line
+	perf_hooks = require('perf_hooks');
+} catch (e) {
+	// node version is too old
+}
+
+const NODEJS_GC_RUNS = 'nodejs_gc_runs';
+const NODEJS_GC_DURATION_SUMMARY_MS = 'nodejs_gc_duration_summary_ms';
+
+function gcKindToString(gcKind) {
+	let gcKindName = '';
+	switch (gcKind) {
+		case perf_hooks.constants.NODE_PERFORMANCE_GC_MAJOR:
+			gcKindName = 'major';
+			break;
+		case perf_hooks.constants.NODE_PERFORMANCE_GC_MINOR:
+			gcKindName = 'minor';
+			break;
+		case perf_hooks.constants.NODE_PERFORMANCE_GC_INCREMENTAL:
+			gcKindName = 'incremental';
+			break;
+		case perf_hooks.constants.NODE_PERFORMANCE_GC_WEAKCB:
+			gcKindName = 'weakcb';
+			break;
+		default:
+			gcKindName = 'unknown';
+			break;
+	}
+	return gcKindName;
+}
+
+module.exports = (registry, config = {}) => {
+	if (!semver.gte(process.version, '8.5.0')) {
+		return () => {};
+	}
```

Comment:
Instead of checking `process.version`, is it sufficient to test `if (!perf_hooks)`? That would get rid of the new dependency too.

Reply:
fixed

Reply:
The counter name should have the total suffix as @zbjornson mentioned

Reply:
I didn't follow too closely, but in #284 was the point that `_total` is an anti-pattern in Prometheus? I only mentioned it here because node-prometheus-gc-stats uses the `_total` suffix.

Reply:
accordingly to promtool check, "_total" suffix is mandatory for counters. So I've added "total" suffix